cmd/kube_create.go

```diff
@@ -0,0 +1,312 @@
+			ZoneID:            zone,
+			ServiceOfferingID: size.ID,
+			TemplateID:        template.ID,
+			RootDiskSize:      10,
+			SecurityGroupIDs:  []egoscale.UUID{*sg.ID},
+		}})
+		if len(errs) > 0 {
+			return errs[0]
+		}
+
+		vm := r[0]
+
+		if _, err := cs.Request(egoscale.CreateTags{
+			ResourceType: vm.ResourceType(),
+			ResourceIDs:  []egoscale.UUID{*vm.ID},
+			Tags:         []egoscale.ResourceTag{{Key: kubeInstanceTagKey, Value: kubeInstanceTagValue}},
+		}); err != nil {
+			return fmt.Errorf("unable to tag cluster instance: %s", err)
+		}
+
+		fmt.Println("Bootstrapping Kubernetes cluster:")
+
+		if err := bootstrapExokubeCluster(&vm, clusterName, kubeCreateDebug); err != nil {
+			return fmt.Errorf("Kubernetes cluster failed: %s", err) // nolint: golint
+		}
+
+		fmt.Printf(`
+Your Kubernetes cluster is ready. What to do now?
+
+1. Install the "kubectl" command, if you don't have it already:
+
+    https://kubernetes.io/docs/tasks/tools/install-kubectl/
+
+2. Execute the following command:
+
+    eval $(exo kube env "%s")
+
+You might want to persist this change by adding it to your shell startup
+configuration (e.g. ~/.bashrc, ~/.zshrc).
+
+3. Check that your cluster is reachable:
+
+    kubectl cluster-info
+
+4. Profit!
+
+5. When you're done with your cluster, you can either stop it using the
+"exo kube stop" command and restart it later using the "exo kube start"
+command, or delete it permanently using the "exo kube delete" command.
+`,
+			clusterName)
+
+		return nil
+	},
+}
+
+// createExokubeSecurityGroup creates the firewall security group to put kube VM instances into, or returns it if it
+// already exists.
+func createExokubeSecurityGroup() (*egoscale.SecurityGroup, error) {
+	var (
+		sg  *egoscale.SecurityGroup
+		err error
+	)
+
+	if sg, err = getSecurityGroupByNameOrID(kubeSecurityGroup); err != nil {
+		if r, ok := err.(*egoscale.ErrorResponse); ok {
+			// Looks like the SG doesn't exist, try to create it
+			if r.ErrorCode == egoscale.ParamError {
+				resp, err := cs.RequestWithContext(context.TODO(), &egoscale.CreateSecurityGroup{
+					Name:        kubeSecurityGroup,
+					Description: "Created by exo CLI",
+				})
+				if err != nil {
+					return nil, err
+				}
+
+				sg = resp.(*egoscale.SecurityGroup)
+
+				sgRules := []egoscale.AuthorizeSecurityGroupIngress{
+					{
+						SecurityGroupID: sg.ID,
+						Description:     "SSH",
+						CIDRList:        []egoscale.CIDR{*egoscale.MustParseCIDR("0.0.0.0/0")},
+						Protocol:        "TCP",
+						StartPort:       22,
+						EndPort:         22,
+					},
+					{
+						SecurityGroupID: sg.ID,
+						Description:     "Kubernetes API",
+						CIDRList:        []egoscale.CIDR{*egoscale.MustParseCIDR("0.0.0.0/0")},
+						Protocol:        "TCP",
+						StartPort:       6443,
+						EndPort:         6443,
+					},
+				}
+
+				for _, rule := range sgRules {
+					if _, err = cs.RequestWithContext(context.TODO(), rule); err != nil {
```

Comment:
You can use `gContext`

Reply:
and can be used to make all request with ctx

Reply:
Done.